server/src/main/java/org/elasticsearch/action/admin/cluster/stats/AnalysisStats.java

```diff
@@ -176,6 +193,28 @@ private static void aggregateAnalysisTypes(
         }
     }
 
+    private static void aggregateSynonymFilterTypes(
+        Collection<Settings> filterSettings,
+        Map<String, IndexFeatureStats> stats,
+        String synonymType,
```

Comment:
I expect synonymType to take different values for the different sources of synonyms (`synonyms_set`, `synonyms_path`, `synonyms`).

Reply:
Should this be an enum instead of a raw string, then, if only certain values are allowed?